src/internal/_objectAssign.js

```diff
@@ -1,24 +1,28 @@
 var _has = require('./_has');
 
-// Based on https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
-module.exports = function _objectAssign(target) {
-  if (target == null) {
-    throw new TypeError('Cannot convert undefined or null to object');
-  }
+module.exports = (function() {
+  return typeof Object.assign === 'function' ?
```

Comment:
does it make sense to do this conditional logic in its own module and then use `internal/_assign` instead?